web/src/search_suggestion.ts

```diff
@@ -982,7 +987,10 @@ export function get_search_result(query_from_pills: string, query_from_text: str
             },
         ];
         attacher.push([...attacher.base, ...suggestion_line]);
-    } else if (all_search_terms.length > 0 && last.operator !== "has" && last.operator !== "is") {
+    } else if (
+        all_search_terms.length > 0 &&
+        !(page_params.is_spectator && ["has", "is"].includes(last.operator))
+    ) {
```

Comment:
What's the story behind the new `is_spectator` check here?

Reply:
I don't completely remember, but I agree it's strange. It's on my list to look into, but I want to start with [this conversation about invalid is](https://chat.zulip.org/#narrow/stream/101-design/topic/showing.20invalid.20is.20pills/near/1839707) since depending on how that goes, this might become irrelevant and removed.